Sources/SwiftlyCore/FileManager+FilePath.swift

```diff
@@ -0,0 +1,179 @@
+import Foundation
+import SystemPackage
+
+public var cwd: FilePath {
+    FileManager.default.currentDir
+}
+
+public var homeDir: FilePath {
+    FileManager.default.homeDir
+}
+
+public var tmpDir: FilePath {
+    FileManager.default.temporaryDir
+}
+
+public func fileExists(atPath: FilePath) async throws -> Bool {
+    FileManager.default.fileExists(atPath: atPath)
+}
+
+public func remove(atPath: FilePath) async throws {
+    try FileManager.default.removeItem(atPath: atPath)
+}
+
+public func move(atPath: FilePath, toPath: FilePath) async throws {
+    try FileManager.default.moveItem(atPath: atPath, toPath: toPath)
+}
+
+public func copy(atPath: FilePath, toPath: FilePath) async throws {
+    try FileManager.default.copyItem(atPath: atPath, toPath: toPath)
+}
+
+public func mkdir(atPath: FilePath, parents: Bool = false) async throws {
+    try FileManager.default.createDir(atPath: atPath, withIntermediateDirectories: parents)
+}
+
+public func cat(atPath: FilePath) async throws -> Data {
+    guard let data = FileManager.default.contents(atPath: atPath) else {
+        throw SwiftlyError(message: "File at path \(atPath) could not be read")
+    }
+
+    return data
+}
+
+public func mktemp(ext: String? = nil) -> FilePath {
+    FileManager.default.temporaryDir.appending("swiftly-\(UUID())\(ext ?? "")")
+}
+
+public func withTemporary<T>(files: FilePath..., f: () async throws -> T) async throws -> T {
+    try await withTemporary(files: files, f: f)
+}
+
+public func withTemporary<T>(files: [FilePath], f: () async throws -> T) async throws -> T {
+    do {
+        let t: T = try await f()
+
+        for f in files {
+            try? await remove(atPath: f)
+        }
+
+        return t
+    } catch {
+        // Sort the list in case there are temporary files nested within other temporary files
+        for f in files.map(\.string).sorted() {
+            try? await remove(atPath: FilePath(f))
+        }
+
+        throw error
+    }
+}
+
+public func create(file: FilePath, contents: Data?, mode: Int = 0) async throws {
+    if mode != 0 {
+        _ = FileManager.default.createFile(atPath: file.string, contents: contents, attributes: [.posixPermissions: mode])
+    } else {
+        _ = FileManager.default.createFile(atPath: file.string, contents: contents)
+    }
+}
+
+public func ls(atPath: FilePath) async throws -> [String] {
+    try FileManager.default.contentsOfDir(atPath: atPath)
+}
+
+public func readlink(atPath: FilePath) async throws -> FilePath {
+    try FileManager.default.destinationOfSymbolicLink(atPath: atPath)
+}
+
+public func symlink(atPath: FilePath, linkPath: FilePath) async throws {
+    try FileManager.default.createSymbolicLink(atPath: atPath, withDestinationPath: linkPath)
+}
+
+public func chmod(atPath: FilePath, mode: Int) async throws {
```

Comment:
Is there a reason these are defined as free functions instead of extensions on FileManager? Seems like some vars like `homeDir` are implemented in two places.

Reply:
Generally I persist a local var like `let fm = FileManager.default` to make these calls more concise (i.e. `fm.chmod(...)`). This also lets you more easily swap out the FileManager implementation in tests to, say, an in memory file system. See something like [FileSystem in tools-support-core](https://github.com/swiftlang/swift-tools-support-core/blob/main/Sources/TSCBasic/FileSystem.swift#L177).